clients/client/src/main/java/org/projectnessie/client/util/v2api/ClientSideGetMultipleNamespaces.java

```diff
@@ -51,9 +51,11 @@ public GetNamespacesResponse get() throws NessieReferenceNotFoundException {
     try {
       GetEntriesBuilder getEntries = api.getEntries().refName(refName).hashOnRef(hashOnRef);
 
-      if (namespace != null) {
+      if (namespace != null && !namespace.isEmpty()) {
+        String nsName = namespace.name();
         getEntries.filter(
-            String.format("entry.key.startsWith('%s')", ContentKey.of(namespace.getElements())));
+            String.format(
+                "entry.encodedKey == '%s' || entry.encodedKey.startsWith('%s.')", nsName, nsName));
```

Comment:
Newer V2 clients will not be able to run this against older V2 servers, but it's probably ok at this stage of API v2 development.

Reply:
yep, should be okay